src/main/kotlin/org/opensearch/integrations/index/IntegrationIndex.kt

```diff
@@ -0,0 +1,132 @@
+            val indexMappingSource = classLoader.getResource(INTEGRATIONS_MAPPING_FILE_NAME)?.readText()!!
+            val indexSettingsSource = classLoader.getResource(INTEGRATIONS_SETTINGS_FILE_NAME)?.readText()!!
+            val request = CreateIndexRequest(INDEX_NAME)
+                .mapping(indexMappingSource, XContentType.YAML)
+                .settings(indexSettingsSource, XContentType.YAML)
+            try {
+                val actionFuture = client.admin().indices().create(request)
+                val response = actionFuture.actionGet(PluginSettings.operationTimeoutMs)
+                if (response.isAcknowledged) {
+                    log.info("$LOG_PREFIX:Index $INDEX_NAME creation Acknowledged")
+                } else {
+                    error("$LOG_PREFIX:Index $INDEX_NAME creation not Acknowledged")
+                }
+                //
+            } catch (exception: ResourceAlreadyExistsException) {
+                log.warn("message: ${exception.message}")
+            } catch (exception: Exception) {
+                if (exception.cause !is ResourceAlreadyExistsException) {
+                    throw exception
+                }
+            }
+            this.mappingsUpdated = true
+        } else if (!this.mappingsUpdated) {
+            updateMappings()
+        }
+    }
+
+    /**
+     * Check if the index is created and available.
+     * @param index
+     * @return true if index is available, false otherwise
+     */
+    private fun isIndexExists(index: String): Boolean {
+        val clusterState = clusterService.state()
+        return clusterState.routingTable.hasIndex(index)
+    }
+
+    /**
+     * Check if the index mappings have changed and if they have, update them
+     */
+    private fun updateMappings() {
+        val classLoader = IntegrationIndex::class.java.classLoader
+        val indexMappingSource = classLoader.getResource(INTEGRATIONS_MAPPING_FILE_NAME)?.readText()!!
+        val request = PutMappingRequest(INDEX_NAME)
+            .source(indexMappingSource, XContentType.YAML)
+        try {
+            val actionFuture = client.admin().indices().putMapping(request)
+            val response = actionFuture.actionGet(PluginSettings.operationTimeoutMs)
+            if (response.isAcknowledged) {
+                log.info("$LOG_PREFIX:Index $INDEX_NAME update mapping Acknowledged")
+            } else {
+                error("$LOG_PREFIX:Index $INDEX_NAME update mapping not Acknowledged")
+            }
+            this.mappingsUpdated = true
+        } catch (exception: IndexNotFoundException) {
+            log.error("$LOG_PREFIX:IndexNotFoundException:", exception)
+        }
+    }
+
+    /**
+     * Create observability object
+     *
+     * @param observabilityObjectDoc
+     * @param id
+     * @return object id if successful, otherwise null
+     */
+    fun createIntegrationObject(integrationObjectDoc: Integration, id: String? = null): String? {
```

Comment:
we also need a delete integration object

Reply:
Out of scope for this PR